llvm/lib/Target/NVPTX/NVPTXSubtarget.cpp

```diff
@@ -55,19 +55,9 @@ NVPTXSubtarget::NVPTXSubtarget(const Triple &TT, const std::string &CPU,
                                const std::string &FS,
                                const NVPTXTargetMachine &TM)
     : NVPTXGenSubtargetInfo(TT, CPU, /*TuneCPU*/ CPU, FS), PTXVersion(0),
-      FullSmVersion(200), SmVersion(getSmVersion()), TM(TM),
+      FullSmVersion(200), SmVersion(getSmVersion()),
       TLInfo(TM, initializeSubtargetDependencies(CPU, FS)) {}
 
-bool NVPTXSubtarget::hasImageHandles() const {
-  // Enable handles for Kepler+, where CUDA supports indirect surfaces and
-  // textures
-  if (TM.getDrvInterface() == NVPTX::CUDA)
```

Comment:
Technically we didn't get rid of sm_20 yet, though it's practical usability is likely close to zero these days.
This may be worth mentioning in release notes as one of the first steps on the way to get rid of pre-kepler stuff.

Reply:
This isn't actually breaking sm_20 (any more than it already is broken). We're just trying to avoid using image handles in all cases after this change (we used to only do this for sm_20), since references are preferable to handles on newer architectures too, even if handles are supported. 

On a side note, I agree we should deprecate and remove support for older architectures. For reference, I think the minimum supported by nvcc is sm_50.

Reply:
> I think the minimum supported by nvcc is sm_50.

The bar here is what hardware is still out there in quantities sufficient to keep support for them around. AWS still seems to list P2 instances with K80: 
![image](https://github.com/user-attachments/assets/ea1d3885-6443-4c90-b50b-fe309af17a6e)

I think it's the current practical low bar on the support.

Reply:
Sounds good, thanks for the explanation of how we're choosing the minimum supported version.